recipes-graphics/mesa/mesa-pvr.inc

```diff
@@ -0,0 +1,391 @@
+SUMMARY = "PVR Mesa fork"
+DESCRIPTION = "Mesa is an open-source implementation of the OpenGL specification - \
+a system for rendering interactive 3D graphics.  \
+A variety of device drivers allows Mesa to be used in many different environments \
+ranging from software emulation to complete hardware acceleration for modern GPUs. \
+Mesa is used as part of the overall Direct Rendering Infrastructure and X.org \
+environment."
+
+HOMEPAGE = "http://mesa3d.org"
+BUGTRACKER = "https://bugs.freedesktop.org"
+SECTION = "x11"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://docs/license.rst;md5=17a4ea65de7a9ab42437f3131e616a7f"
+
+PE = "2"
+
+SRC_URI = "https://mesa.freedesktop.org/archive/mesa-${PV}.tar.xz \
```

Comment:
You should be able to use a `bbappend` to apply the patches on top of the existing mesa recipe. That would reduce this file to pretty much just a list of patches in `SRC_URI`.

In saying that, it's a lot of patches. Maybe we should hold the mesa support off until some more of these are merged upstream

Reply:
From the listed patches I think that I will probably only need the top-most two, which is the next step I want to test. Yet the Starfive Mesa is quite old in comparison to Mesa upstream, in particularly in regard to some Mesa restructuring around DRI drivers, which is some non trivial porting. Hope to tell more at the end of the week (and hopefully will drop 90% of the patches).
Regarding the Mesa fork, I am not sure yet if a machine specific bbappend is cleaner, mostly because I have to force binary GLESv2 with Mesa's EGL, which the sanity checks in the OE Mesa recipe try to prevent.

Reply:
please see my longer comment from today in the MR discussion. tl;dr: this patch set only works with the Mesa Amber branch AFAIS and I do not see the Amber branch packaged anywhere in OE.
So for now, I think sticking with this fork is the most viable solution. Rebasing it to a more recent Amber branch version I would do in another MR which would also drop a few patches. Yet, I do not expect most of the VisionFive2 driver specific patches ever to be backported to Mesa. The more realistic scenario IMO is that we earlier see an official driver (not that I had any information about, just guessing from Imagination's press release a year ago!)